lib/Parse/Lexer.cpp

```diff
@@ -1890,7 +1890,7 @@ void Lexer::lexStringLiteral(unsigned CustomDelimiterLen) {
         diagnose(--TmpPtr, diag::opening_paren);
 
         return formToken(tok::unknown, TokStart);
-      } else if (*CurPtr == '\r' || *CurPtr == '\n') {
+      } else if (*CurPtr == '\r' || *CurPtr == '\n' || *CurPtr == BufferEnd) {
```

Comment:
[`BufferEnd`](https://github.com/apple/swift/blob/55fb3d1ff9b2283db61f53004e540553379221fa/include/swift/Parse/Lexer.h#L102) is a pointer, no need to dereference.

Reply:
Oh didn't check for that, thanks for letting me know